daemon/src/maker_cfd_actor.rs

```diff
@@ -300,7 +298,107 @@ impl MakerCfdActor {
 
         Ok(())
     }
+
+    async fn handle_accept_order(&mut self, msg: AcceptOrder) -> Result<()> {
+        tracing::debug!(%msg.order_id, "Maker accepts an order" );
+
+        let mut conn = self.db.acquire().await?;
+
+        // Validate if order is still valid
+        let cfd = load_cfd_by_order_id(msg.order_id, &mut conn).await?;
+
+        let taker_id = match cfd {
+            Cfd {
+                state: CfdState::IncomingOrderRequest { taker_id, .. },
+                ..
+            } => taker_id,
+            _ => {
+                tracing::warn!("Order is in invalid state. Ignoring trying to accept it.");
+                anyhow::bail!("Order is in invalid state. Ignoring trying to accept it.")
```

Comment:
Just the `bail!` should be sufficient because of the `log_error` macro. This way we are going to see the message twice!